oneanddone/tasks/static/js/tasklist.js

```diff
@@ -0,0 +1,415 @@
+      }.bind(this)
+    });
+  },
+  loadMoreTasks: function() {
+    let filters = this.state.filters;
+    filters.page ++;
+    this.setState({filters: filters});
+    this.loadTasksFromServer(true);
+  },
+  getInitialState: function() {
+    return {
+      data: [],
+      filters: {search: '', execution_time: [], team: [], project: [], type: [], page: 1},
+      moreTasks: false,
+      teams: [],
+      projects: [],
+      types: []
+    };
+  },
+  componentDidMount: function() {
+    this.loadTasksFromServer();
+    this.loadTeamsFromServer();
+    this.loadProjectsFromServer();
+    this.loadTypesFromServer();
+  },
+  render: function() {
+    return (
+      <div className="taskPage">
+        <TaskFilters filters={this.state.filters}
+                     onFilterChange={this.loadTasksFromServer}
+                     teams={this.state.teams}
+                     types={this.state.types}
+                     projects={this.state.projects}/>
+        <TaskList data={this.state.data} onLoadMore={this.loadMoreTasks} moreTasks={this.state.moreTasks} />
+      </div>
+    );
+  }
+});
+
+let TaskFilters = React.createClass({
+  handleSearchChange: function(e) {
+    this.props.filters.search = e.target.value;
+    this.props.onFilterChange();
+  },
+  render: function() {
+    return (
+      <div className="task-filters">
+        <form method="get" className="filters">
+          <fieldset>
+            <SearchFilter {...this.props} />
+            <ETFilter {...this.props} />
+            <TeamFilter {...this.props} />
+            <ProjectFilter {...this.props} />
+            <TypeFilter {...this.props} />
+          </fieldset>
+        </form>
+      </div>
+    );
+  }
+});
+
+let SearchFilter = React.createClass({
+  handleChange: function(e) {
+    this.props.filters.search = e.target.value;
+    this.props.onFilterChange();
+  },
+  render: function() {
+    return (
+      <p>
+        <input type="text" id="search" placeholder="Search for tasks" onChange={this.handleChange} />
+      </p>
+    );
+  }
+});
+
+let ETFilter = React.createClass({
+  render: function() {
+    let etNodes = [15, 30, 45, 60].map(function(executionTime) {
+      return (
+        <ET executionTime={executionTime}
+            key={executionTime}
+            {...this.props} />
+      );
+    }, this);
+    return (
+      <div className="et-filter">
+        <p>Estimated minutes to complete:</p>
+        <p>
+          {etNodes}
+        </p>
+      </div>
+    );
+  }
+});
+
+let ET = React.createClass({
+  handleChange: function(e) {
+    let executionTime = this.props.filters.execution_time;
+    if (e.target.checked) {
+      executionTime.push(e.target.value);
+    } else {
+      executionTime.splice($.inArray(e.target.value, executionTime), 1);
+    }
+    this.props.onFilterChange();
+  },
+  render: function() {
+    let etId = 'et' + this.props.executionTime;
+    return (
+      <label htmlFor={etId}>
```

Comment:
You may want to add a bit of padding here. I was hacking the dev tools and adding a 10-15px of `padding-right` seemed to space these out a bit.